### PR TITLE
CLU build sdk

### PR DESCRIPTION
**Pull request by mshaban-msft**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/cognitivelanguage/azure-ai-language-conversations/swagger/README.md [hunk 1]
```diff
@@ -20,84 +20,45 @@ autorest
...
 ```
 
-### Rename 'confidenceScore' in Qna intent result
+### Rename 'duplicate schema' errors in 'Job State'
 
 ```yaml
 directive:
     - from: swagger-document
-      where: $["definitions"]["KnowledgeBaseAnswer"]["properties"]["confidenceScore"]
+      where: $["definitions"]["JobState"]["properties"]
```
Comment (heaths) [inline sdk/cognitivelanguage/azure-ai-language-conversations/swagger/README.md:61]: These really should be in the swagger. Lots of toosl are affected. There is a push to set `x-ms-client-name` to disambiguate. /cc @abhahn

File: sdk/cognitivelanguage/azure-ai-language-conversations/swagger/README.md [hunk 2]
```diff
@@ -20,84 +20,45 @@ autorest
 ### Settings
 
 ```yaml
-input-file: https://raw.githubusercontent.com/Azure/azure-rest-api-specs/main/specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json
+input-file: https://raw.githubusercontent.com/Azure/azure-rest-api-specs/release-cognitiveservices-Language-2022-05-01-preview/specification/cognitiveservices/data-plane/Language/preview/2022-05-15-preview/analyzeconversations.json
```
Comment (heaths) [inline sdk/cognitivelanguage/azure-ai-language-conversations/swagger/README.md:23]: You should actually use a commit so it doesn't change unexpectedly - especially for releases.

**Review by heaths — CHANGES_REQUESTED** [review]

File: sdk/cognitivelanguage/azure-ai-language-conversations/swagger/README.md [hunk 3]
```diff
@@ -165,3 +165,23 @@ directive:
...
+        $["x-ms-discriminator-value"] = "IssueResolutionSummarization";
+```
+
+```yaml
+directive:
+    - from: swagger-document
+      where: $["definitions"]["AnalyzeConversationPIITask"]
+      transform: >
+        $["x-ms-discriminator-value"] = "ConversationPII";
+```
```
Comment (heaths) [inline sdk/cognitivelanguage/azure-ai-language-conversations/swagger/README.md:187]: Tip: you can declutter like so:
```suggestion
```yaml
directive:
- from: swagger-document
  where: $["definitions"]["AnalyzeConversationSummarizationTask"]
  transform: >
    $["x-ms-discriminator-value"] = "IssueResolutionSummarization";

- from: swagger-document
  where: $["definitions"]["AnalyzeConversationPIITask"]
  transform: >
    $["x-ms-discriminator-value"] = "ConversationPII";
```

It's a YAML array so you can put it under one directive for brevity. You can also elide the whitespace on

Comment (heaths) [reply sdk/cognitivelanguage/azure-ai-language-conversations/swagger/README.md:187]: Separately, can we rename these using `x-ms-client-name` in the swagger so all autorest-generated clients have these changes? Or, at the very least, put the transforms in the readme.md in the swagger repo. It's best to avoid per-repo transforms if possible.

Comment (mshaban-msft) [reply sdk/cognitivelanguage/azure-ai-language-conversations/swagger/README.md:202]: yeah sure i'm working on refactoring the directives file now

File: sdk/cognitivelanguage/azure-ai-language-conversations/azure/ai/language/conversations/_operations/_operations.py [hunk 4]
```diff
@@ -122,3 +158,141 @@ def analyze_conversation(
...
+            return cls(pipeline_response, deserialized, response_headers)
+
+        return deserialized
+
+
+
+    @distributed_trace
+    def begin_submit_conversation_job(
+        self,
+        body: _models.AnalyzeConversationJobsInput,
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-conversations/azure/ai/language/conversations/_operations/_operations.py:223]: Let's rename the parameter to something like `jobs`

File: sdk/cognitivelanguage/azure-ai-language-conversations/swagger/README.md [hunk 5]
```diff
@@ -29,36 +29,86 @@ clear-output-folder: true
 no-namespace-folders: true
 python: true
 title: ConversationAnalysisClient
+tag: release_2022_05_01_preview
+openapi-type: data-plane
 version-tolerant: true
 models-mode: msrest
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-conversations/swagger/README.md:35]: Could we please remove this `models-mode: msrest` line?
I believe we're going to GA this release without models, so we should start with this beta release.

File: sdk/cognitivelanguage/azure-ai-language-conversations/swagger/README.md [hunk 6]
```diff
@@ -94,10 +144,207 @@ directive:
...
 ```
 
+### Fix `enum` error
+
 ```yaml
 directive:
   - from: swagger-document
     where: $["definitions"]["CustomConversationTaskParameters"]
     transform: >
         delete $.properties["stringIndexType"]
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-conversations/swagger/README.md:154]: I think in QnA we had to compensate for deleting this property by patching the client to set it explicitly. Could we copy that logic from the questionanswering SDK to apply it here?
Here's the patch:
https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/models/_patch.py#L68-L70

File: sdk/cognitivelanguage/azure-ai-language-conversations/azure/ai/language/conversations/_operations/_operations.py [hunk 7]
```diff
@@ -122,3 +158,141 @@ def analyze_conversation(
...
+
+        if cls:
+            return cls(pipeline_response, deserialized, response_headers)
+
+        return deserialized
+
+
+
+    @distributed_trace
+    def begin_submit_conversation_job(
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-conversations/azure/ai/language/conversations/_operations/_operations.py:221]: I wonder if we should rename this operation to just `begin_conversation_analysis` or `begin_conversation_job`?
@heaths what do you think of this API name currently?

File: sdk/cognitivelanguage/azure-ai-language-conversations/samples/reslotion_and_pii/sample_conv_pii_transcript_input.py [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+                            pii_categories=["all"]
+                        )
+                    )
+                ]
+            )
+        )
+
+        # view result
+        result = poller.result()
+        task_result = result.tasks.items[0]
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-conversations/samples/reslotion_and_pii/sample_conv_pii_transcript_input.py:105]: Should the sample loop over all the task items?

File: sdk/cognitivelanguage/azure-ai-language-conversations/samples/reslotion_and_pii/sample_conv_pii_transcript_input_dict_parms.py [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+    endpoint = os.environ["AZURE_CONVERSATIONS_ENDPOINT"]
+    key = os.environ["AZURE_CONVERSATIONS_KEY"]
+
+    # analyze quey
+    client = ConversationAnalysisClient(endpoint, AzureKeyCredential(key))
+    with client:
+
+        poller = client.begin_submit_conversation_job(
+            body={
+                "displayName": "Analyze conversations from xxx",
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-conversations/samples/reslotion_and_pii/sample_conv_pii_transcript_input_dict_parms.py:44]: Maybe something like "Analyze PII in conversation"?

File: sdk/cognitivelanguage/azure-ai-language-conversations/samples/reslotion_and_pii/sample_conv_pii_transcript_input.py [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+
+    Set the environment variables with your own values before running the sample:
+    1) AZURE_CONVERSATIONS_ENDPOINT                       - endpoint for your CLU resource.
+    2) AZURE_CONVERSATIONS_KEY                            - API key for your CLU resource.
+"""
+
+from azure.ai.language import conversations
+
+
+def sample_conv_pii_transcript_input():
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-conversations/samples/reslotion_and_pii/sample_conv_pii_transcript_input.py:26]: Could we add async versions? :)

File: sdk/cognitivelanguage/azure-ai-language-conversations/azure/ai/language/conversations/_operations/_operations.py (hunk 7 above)
Comment (heaths) [reply sdk/cognitivelanguage/azure-ai-language-conversations/azure/ai/language/conversations/_operations/_operations.py:221]: I don't think we typically want to mention "job" or "task". It's typically that I've seen. Probably just `begin_conversation_analysis` as you suggested.

File: sdk/cognitivelanguage/azure-ai-language-conversations/samples/reslotion_and_pii/sample_conv_pii_transcript_input.py (hunk 10 above)
Comment (mshaban-msft) [reply sdk/cognitivelanguage/azure-ai-language-conversations/samples/reslotion_and_pii/sample_conv_pii_transcript_input.py]: i'm just waiting for final fixes to service apis in order to make sure no more changes in sync samples/tests

File: sdk/cognitivelanguage/azure-ai-language-conversations/samples/reslotion_and_pii/sample_conv_pii_transcript_input_dict_parms.py (hunk 9 above)
Comment (mshaban-msft) [reply sdk/cognitivelanguage/azure-ai-language-conversations/samples/reslotion_and_pii/sample_conv_pii_transcript_input_dict_parms.py]: i don't agree with this
simply because in the async api, we can submit either `issue resolution` or `conv pii` tasks
but weirdly enough, autorest appends `begin` in front of the async operation after rename
do you have an idea how to fix that?

File: sdk/cognitivelanguage/azure-ai-language-conversations/samples/reslotion_and_pii/sample_conv_pii_transcript_input.py (hunk 8 above)
Comment (mshaban-msft) [reply sdk/cognitivelanguage/azure-ai-language-conversations/samples/reslotion_and_pii/sample_conv_pii_transcript_input.py]: yes! but in our case we submitted just one conversation item

File: sdk/cognitivelanguage/azure-ai-language-conversations/azure/ai/language/conversations/_operations/_operations.py (hunk 7 above)
Comment (mshaban-msft) [reply sdk/cognitivelanguage/azure-ai-language-conversations/azure/ai/language/conversations/_operations/_operations.py:221]: renamed to `begin_conversation_analysis`
but it's a little confusing having
- sync operation as `analyze_conversation`
- async operation as `begin_conversation_analysis`

File: sdk/cognitivelanguage/azure-ai-language-conversations/swagger/README.md (hunk 5 above)
Comment (mshaban-msft) [reply sdk/cognitivelanguage/azure-ai-language-conversations/swagger/README.md:35]: that means there won't be any models, right?!
i tried removing this, and all operations accept json formats input/output
i this normal?

File: sdk/cognitivelanguage/azure-ai-language-conversations/swagger/README.md (hunk 3 above)
Comment (mshaban-msft) [reply sdk/cognitivelanguage/azure-ai-language-conversations/swagger/README.md:209]: actually, after speaking with the service people, they told me that APIM/service changes will catch up to swagger definitions which means this entire section named `Fix Swagger/API mismatch errors` will be removed from swagger directives.

File: sdk/cognitivelanguage/azure-ai-language-conversations/swagger/README.md (hunk 2 above)
Comment (mshaban-msft) [reply sdk/cognitivelanguage/azure-ai-language-conversations/swagger/README.md]: will do
i'm just using my fork which have some changes i couldn't do using directives in swagger

File: sdk/cognitivelanguage/azure-ai-language-conversations/swagger/README.md (hunk 6 above)
Comment (mshaban-msft) [reply sdk/cognitivelanguage/azure-ai-language-conversations/swagger/README.md:154]: good catch!

File: sdk/cognitivelanguage/azure-ai-language-conversations/azure/ai/language/conversations/_operations/_operations.py (hunk 7 above)
Comment (heaths) [reply sdk/cognitivelanguage/azure-ai-language-conversations/azure/ai/language/conversations/_operations/_operations.py]: That's a good point. I would bring it up with @johanste and @tg-msft.

File: sdk/cognitivelanguage/azure-ai-language-conversations/azure/ai/language/conversations/_client.py [hunk 11]
```diff
@@ -7,35 +7,49 @@
...
-
+    upstream service. The asynchronous APIs in this suite enable tasks like Conversation
+    Summarization and Conversational PII detection.
+
+    :ivar conversation_analysis: ConversationAnalysisOperations operations
+    :vartype conversation_analysis:
+     azure.ai.language.conversations.operations.ConversationAnalysisOperations
+    :ivar analyze_conversation: AnalyzeConversationOperations operations
+    :vartype analyze_conversation:
+     azure.ai.language.conversations.operations.AnalyzeConversationOperations
```
Comment (kristapratico) [inline sdk/cognitivelanguage/azure-ai-language-conversations/azure/ai/language/conversations/_client.py:42]: Hm I think we want all the operations directly accessible via the client without needing to do something like `client.operation_group.method()`

edit: If I edit the swagger README, this seems achievable. Here's what I used: https://gist.github.com/kristapratico/0a5f898faf3becfa64872ab331b3b751#rename-client-operations

File: sdk/cognitivelanguage/azure-ai-language-conversations/azure/ai/language/conversations/aio/operations/_operations.py [hunk 12]
```diff
@@ -0,0 +1,516 @@
...
+                # JSON input template you can fill out and use as your body input.
+                task = {
+                    kind: kind
+                }
+
+                # response body for status code(s): 200
+                response.json() == {
+                    kind: kind
+                }
+        """
```
Comment (kristapratico) [inline sdk/cognitivelanguage/azure-ai-language-conversations/azure/ai/language/conversations/aio/operations/_operations.py:77]: This generated JSON input doesn't include all the info to create a request... I think we'll want to _patch this one so it has the correct copy/paste for input and a good example response. We can do this in another PR, though, after this merges.

File: sdk/cognitivelanguage/azure-ai-language-conversations/azure/ai/language/conversations/aio/operations/_operations.py [hunk 13]
```diff
@@ -0,0 +1,516 @@
...
+                        "conversations": [
+                            {
+                                "domain": "str",  # Optional. Enumeration of
+                                  supported conversational domains. Known values are: "finance",
+                                  "healthcare", "generic".
+                                "id": "str",  # Required. Unique identifier for the
+                                  conversation.
+                                "language": "str",  # Required. The language of the
+                                  conversation item in BCP-47 format.
+                                modality: modality
```
Comment (kristapratico) [inline sdk/cognitivelanguage/azure-ai-language-conversations/azure/ai/language/conversations/aio/operations/_operations.py:228]: Something happened here as well... I think `modality` breaks this from being copy/paste-able. @iscai-msft

File: sdk/cognitivelanguage/azure-ai-language-conversations/samples/async/sample_analyze_conversation_app_async.py [hunk 14]
```diff
@@ -49,53 +42,60 @@ async def sample_analyze_conversation_app_async():
...
-                print("extra info")
-                for data in entity.extra_information:
-                    print("kind: {}".format(data.extra_information_kind))
-                    if data.extra_information_kind == "ListKey":
-                        print("key: {}".format(data.key))
-                    if data.extra_information_kind == "EntitySubtype":
-                        print("value: {}".format(data.value))
-
-    # [END analyze_conversation_app_async]
+        result = await client.conversation_analysis.analyze_conversation(
```
Comment (kristapratico) [inline sdk/cognitivelanguage/azure-ai-language-conversations/samples/async/sample_analyze_conversation_app_async.py:45]: I'm hoping that once we regenerate then this will just become `client.analyze_conversation(...)` but let me know if there is still an issue

File: sdk/cognitivelanguage/azure-ai-language-conversations/swagger/README.md (hunk 6 above)
Comment (mshaban-msft) [reply sdk/cognitivelanguage/azure-ai-language-conversations/swagger/README.md:153]: not valid as we're not generating enums anymore

File: sdk/cognitivelanguage/azure-ai-language-conversations/azure/ai/language/conversations/_client.py (hunk 11 above)
Comment (mshaban-msft) [reply sdk/cognitivelanguage/azure-ai-language-conversations/azure/ai/language/conversations/_client.py]: i created this autorest directive to rename operation in the last release
but weirdly enough, when calling the method, client throws an error `'AnalyzeConversationOperations' object is not callable`
just added an 's' to have it as analyze_conversations, and it worked

File: sdk/cognitivelanguage/azure-ai-language-conversations/CHANGELOG.md [hunk 15]
```diff
@@ -1,5 +1,16 @@
 # Release History
 
+## 1.0.0b4 (2022-05-15)
+
+### Features Added
+* Async conversation issue summarization task
+* Async conversation PII extraction task
```
[the remark this answers is not in the document]
Comment (mshaban-msft) [reply sdk/cognitivelanguage/azure-ai-language-conversations/CHANGELOG.md]: the `cancel` operation was added recently (2 days ago)
which we didn't agree on with the team
what do you think we should do?

File: sdk/cognitivelanguage/azure-ai-language-conversations/CHANGELOG.md [hunk 16]
```diff
@@ -1,5 +1,16 @@
 # Release History
 
+## 1.0.0b4 (2022-05-15)
```
[the remark this answers is not in the document]
Comment (mshaban-msft) [reply sdk/cognitivelanguage/azure-ai-language-conversations/CHANGELOG.md:3]: i think we can do this next week

**Review by kristapratico — APPROVED** [review]
Just a few last nits. Looking good!

File: sdk/cognitivelanguage/azure-ai-language-conversations/CHANGELOG.md [hunk 17]
```diff
@@ -1,5 +1,16 @@
...
+
+### Features Added
+* Conversation issue summarization task (Long-running operation)
+* Conversation PII extraction task (Long-running operation)
+
+### Breaking Changes
+* Client now uses python dictionaries for method parameters and results instead of classes.
+* Many input and result parameter name changes in `analyze_conversations()` method
```
[the remark this answers is not in the document]
Comment (mshaban-msft) [reply sdk/cognitivelanguage/azure-ai-language-conversations/CHANGELOG.md:11]: yeah! there was a bug introduced because of the newly added `cancel` api
i couldn't name the method in singular form, but i fixed that
will update changelog

File: sdk/cognitivelanguage/azure-ai-language-conversations/README.md [hunk 18]
```diff
@@ -19,7 +20,7 @@ _Azure SDK Python packages support for Python 2.7 ended 01 January 2022. For mor
 
 * Python 3.6 or later is required to use this package.
 * An [Azure subscription][azure_subscription]
-* An existing Text Analytics resource
+* An existing Azure Language Service Resource
 
 > Note: the new unified Cognitive Language Services are not currently available for deployment.
```
[the remark this answers is not in the document]
Comment (mshaban-msft) [reply sdk/cognitivelanguage/azure-ai-language-conversations/README.md:25]: why? users still need to create a Language Resource

[the remark this answers is not in the document]
Comment (kristapratico) [reply sdk/cognitivelanguage/azure-ai-language-conversations/README.md:25]: Yes, but this note on L25 says that the language service is not currently available for deployment. That is not true -- you can create/deploy a Language service resource.

File: sdk/cognitivelanguage/azure-ai-language-conversations/swagger/README.md [hunk 19]
```diff
@@ -20,7 +20,7 @@ autorest
 ### Settings
 
 ```yaml
-input-file: https://raw.githubusercontent.com/Azure/azure-rest-api-specs/main/specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json
+input-file: https://raw.githubusercontent.com/Azure/azure-rest-api-specs/d1716d13b0814a9d0785eda9a74529a315212f53/specification/cognitiveservices/data-plane/Language/preview/2022-05-15-preview/analyzeconversations.json
```
Comment (heaths) [inline sdk/cognitivelanguage/azure-ai-language-conversations/swagger/README.md:23]: This will need to change once I merge https://github.com/Azure/azure-rest-api-specs/pull/19186

File: sdk/cognitivelanguage/azure-ai-language-conversations/swagger/README.md [hunk 20]
```diff
@@ -29,26 +29,85 @@ clear-output-folder: true
 no-namespace-folders: true
 python: true
 title: ConversationAnalysisClient
+tag: release_2022_05_01_preview
+openapi-type: data-plane
 version-tolerant: true
-models-mode: msrest
-package-version: 1.0.0b3
+package-version: 1.0.0b4
```
Comment (heaths) [inline sdk/cognitivelanguage/azure-ai-language-conversations/swagger/README.md:35]: 1.1.0b1

**Review by heaths — CHANGES_REQUESTED** [review]

**Review by heaths — APPROVED** [review]
Code structure/docs LGTM
